src/core/src/main/java/org/apache/jmeter/save/CSVSaveService.java

```diff
@@ -786,7 +785,13 @@ private void addDelim() {
         // quotes:
         public void append(String s) {
             addDelim();
-            sb.append(quoteDelimiters(s, specials));
+            sb.append(escapeControlChars(quoteDelimiters(s, specials)));
+        }
+
+        private static String escapeControlChars(String value) {
+            return value
+                    .replace("\n", "\\n")
+                    .replace("\r", "\\r");
```

Comment:
This looks strange as `\n` and `\r` are already a part of `specials`, so they should already be replaced